### PR TITLE
stm32mp1: fixes for stm32_i2c support

### PR DESCRIPTION
**Pull request by etienne-lms**
This series fixes bugs in I2C support in the platform.

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/conf.mk [hunk 1]
```diff
@@ -35,9 +35,15 @@ CFG_WITH_LPAE ?= y
 CFG_WITH_STACK_CANARIES ?= y
 CFG_MMAP_REGIONS ?= 23
 
+ifeq ($(CFG_EMBED_DTB_SOURCE_FILE),)
+# Some drivers mandate DT support
```
Comment (jbech-linaro) [inline core/arch/arm/plat-stm32mp1/conf.mk]: Is it DT who require I2C or I2C who requires DT? I get puzzled when reading the comment and seeing the code.

Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/conf.mk]: I2C requires DT. Main concern is I2C configuration that mandate quite a lot of settings which we to decide to support only using DT means. If embedded DT is disabled, I2C driver is forced disabled.

**Comment by etienne-lms** [comment]
There's a checkpatch issue. I'll fix when applying r-b tag.

**Comment by etienne-lms** [comment]
tag applied.

> Is it DT who require I2C or I2C who requires DT? I get puzzled when reading the comment and seeing the code.

Reading back code and commit message, I found it quite clear. Please tell if it is not the case so that I maybe update platform `conf.mk` or commit message to better reflect the design.

**Comment by jbech-linaro** [comment]
> Reading back code and commit message, I found it quite clear. Please tell if it is not the case so that I maybe update platform `conf.mk` or commit message to better reflect the design.

It works as it is, I get the message (even though I think it's a little bit backwards). I.e., no need to change anything from my point of view.

**Pull request merged**